### PR TITLE
fix tag assignment when copying

### PR DESCRIPTION
**Pull request by RaananW**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

**Review by Popov72 — APPROVED** [review]

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13555/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13555/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13555/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13555/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13555/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13555/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13555/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13555/merge#BCU1XR#0

File: packages/dev/core/src/Misc/decorators.ts [hunk 1]
```diff
@@ -22,8 +22,8 @@ const _copySource = function <T>(creationFunction: () => T, source: T, instancia
     const destination = creationFunction();
 
     // Tags
-    if (Tags) {
-        Tags.AddTagsTo(destination, (<any>source).tags);
+    if (Tags && Tags.HasTags(source)) {
```
Comment (RaananW) [inline packages/dev/core/src/Misc/decorators.ts:25]:
```suggestion
    if (Tags && Tags.HasTags(source)) {
```

**Comment by RaananW** [comment]
CI passed - https://dev.azure.com/babylonjs/ContinousIntegration/_build/results?buildId=24249&view=results but not reported here for some reason. Merging.

**Pull request merged**